### PR TITLE
support for advanced pinot table config

### PR DESCRIPTION
**Pull request by laxmanchekka**
## Description
Support advanced table config (startree config, segment partition config, routing config)

### Testing
Added unit tests. 

### Checklist:
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been merged and published in downstream modules

### Documentation
NA

### CODE REVIEW

File: view-creator-framework/build.gradle.kts [hunk 1]
```diff
@@ -85,4 +76,9 @@ dependencies {
   testImplementation("org.mockito:mockito-core:4.5.1")
 }
 
+// Disabling compatibility check for the test avro definitions.
+tasks.named<org.hypertrace.gradle.avro.CheckAvroCompatibility>("avroCompatibilityCheck") {
```
Comment (kotharironak) [inline view-creator-framework/build.gradle.kts:80]: Is this apply only to test?

File: view-creator-framework/src/test/resources/sample-view-generation-spec.conf [hunk 2]
```diff
@@ -72,6 +73,33 @@ pinotRealtime = {
     storageType = "pinot_server"
     serverTag = "tier-for-hot-data"
   }]
+
+  starTreeIndexConfigs = [{
+    dimensionsSplitOrder = [
+      "tenant_id",
+      "start_time_millis"
+    ],
+    skipStarNodeCreationForDimensions = [],
```
Comment (kotharironak) [inline view-creator-framework/src/test/resources/sample-view-generation-spec.conf:82]: If we give `dimensionsSplitOrder`, do we need this `skipStarNodeCreationForDimensions`?

File: view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/pinot/PinotUtils.java [hunk 3]
```diff
@@ -356,7 +370,64 @@ public static TableConfig buildPinotTableConfig(
...
+  }
+
+  private static SegmentPartitionConfig toSegmentPartitionConfig(Config segmentPartitionConfig) {
+    if (segmentPartitionConfig == null) {
+      return null;
+    }
+    Config columnPartitionMap = segmentPartitionConfig.getConfig("columnPartitionMap");
+    Map<String, ColumnPartitionConfig> tableColumnPartitionMap = Maps.newHashMap();
+    for (Entry<String, ConfigValue> entry : columnPartitionMap.entrySet()) {
+      String columnName = ConfigUtil.splitPath(entry.getKey()).get(0);
```
Comment (kotharironak) [inline view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/pinot/PinotUtils.java:395]: Do we have an example of this config? Just curious, why do we have to extract columnName via splitPath

File: view-creator-framework/build.gradle.kts (hunk 1 above)
Comment (laxmanchekka) [reply view-creator-framework/build.gradle.kts:80]: we have only test avdl generation here in this module.

File: view-creator-framework/src/test/resources/sample-view-generation-spec.conf (hunk 2 above)
Comment (laxmanchekka) [reply view-creator-framework/src/test/resources/sample-view-generation-spec.conf:82]: both are two different configurations. `skipStarNodeCreationForDimensions` - can be enabled for dimensions where we don't need star node, say tenant_id in our case.

Comment (laxmanchekka) [reply view-creator-framework/src/test/resources/sample-view-generation-spec.conf:82]: pinot docs for reference
https://docs.pinot.apache.org/basics/indexing/star-tree-index#index-generation

I can explain if not clear.

File: view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/pinot/PinotUtils.java (hunk 3 above)
Comment (laxmanchekka) [reply view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/pinot/PinotUtils.java:395]: splitting is needed to extract the dynamic keys. this map (`tableColumnPartitionMap `) is expected while building table config

Comment (laxmanchekka) [reply view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/pinot/PinotUtils.java:395]: example config is part of the PR itself. check the test config added in PR.

**Comment by laxmanchekka** [comment]
Handling snyk issue requires pinot client upgrade and that's blocked on incompatibility issues with boolean fields.
Need to handle it separately.

File: view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/pinot/PinotUtils.java (hunk 3 above)
Comment (kotharironak) [reply view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/pinot/PinotUtils.java:395]: This is the one, right?

```
segmentPartitionConfig = {
    columnPartitionMap = {
      tenant_id = {
        functionName = "Modulo",
        numPartitions = 4
      }
    }
  }
```

**Review by kotharironak — APPROVED** [review]

**Comment by kotharironak** [comment]
@laxmanchekka Is it possible to take care snyk failure too?

```
Upgrade org.apache.pinot:pinot-tools@0.7.1 to org.apache.pinot:pinot-tools@0.10.0 to fix
  ✗ Deserialization of Untrusted Data [Medium Severity][https://snyk.io/vuln/SNYK-JAVA-COMMONSCOLLECTIONS-472711] in commons-collections:commons-collections@3.2.1
    introduced by org.apache.pinot:pinot-tools@0.7.1 > org.apache.pinot:pinot-core@0.7.1 > commons-collections:commons-collections@3.2.1 and 1 other path(s)
  ✗ Deserialization of Untrusted Data [Critical Severity][https://snyk.io/vuln/SNYK-JAVA-COMMONSCOLLECTIONS-[30](https://github.com/hypertrace/view-generator-framework/runs/6960152129?check_suite_focus=true#step:4:31)078] in commons-collections:commons-collections@3.2.1
    introduced by org.apache.pinot:pinot-tools@0.7.1 > org.apache.pinot:pinot-core@0.7.1 > commons-collections:commons-collections@3.2.1 and 1 other path(s)
```

File: view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/pinot/PinotUtils.java (hunk 3 above)
Comment (laxmanchekka) [reply view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/pinot/PinotUtils.java:395]: yes

**Pull request merged**